Plugin/Catalog/Helper/Product/Configuration/AddVisibleInCheckoutAttributesToCustomOptionsPlugin.php

```diff
@@ -42,8 +43,34 @@ public function __construct(GetVisibleCheckoutAttributesServiceInterface $getVis
     public function afterGetCustomOptions(Configuration $configuration, array $customOptions, ItemInterface $item)
     {
         $attributes = $this->getVisibleCheckoutAttributesService->execute();
+
+        $configurableAttributes = $item->getOptionByCode('attributes') ? $item->getOptionByCode('attributes')->getValue() : [];
+        $configurableAttributes = $configurableAttributes ? array_keys(json_decode($configurableAttributes, true)) : [];
```

Comment:
I did not check how this is encoded, but most probably, we should use `\Magento\Framework\Serialize\Serializer\Json` instead of directly using `json_decode`?